src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -3163,6 +3197,24 @@ private boolean writeBatchData(TDSWriter tdsWriter) throws SQLServerException {
             // No more data available, return false so we do not execute any more batches.
             if (!goToNextRow())
                 return false;
+            
+            
+            if (insertRowByRow) {
+                // read response gotten from goToNextRow()
+                ((SQLServerResultSet) sourceResultSet).getTDSReader().readPacket();
+
+                // Create and send the initial command for bulk copy ("INSERT BULK ...").
+                tdsWriter = command.startRequest(TDS.PKT_QUERY);
+                String bulkCmd = createInsertBulkCommand(tdsWriter);
+                tdsWriter.writeString(bulkCmd);
```

Comment:
sending 'bulkCmd' in loop is fine, but driver will generate the same `insert bulk .....` command again and again in line 3208.